test/e2e/reconciler/cluster/controller_test.go

```diff
@@ -272,26 +269,11 @@ func ExpectCowboys(ctx context.Context, t framework.TestingTInterface, client wi
 	}
 	return func(seed *wildwestv1alpha1.Cowboy, expectation CowboyExpectation) error {
 		return expecter.ExpectBefore(ctx, func(ctx context.Context) (done bool, err error) {
-			// we are using a seed from one kcp to expect something about an object in
-			// another kcp, so the cluster names will not match - this is fine, just do
-			// client-side filtering for what we know
-			all, err := informer.Lister().Cowboys(seed.Namespace).List(labels.Everything())
+			cowboy, err := informer.Lister().Cowboys(seed.Namespace).Get(seed.Name)
```

Comment:
The lister requires a cluster-aware string on GET - e.g. cluster + name. We don't know the cluster, or we don't care. So we list and filter.

Reply:
thanks. That explain it.